Snowflake.Data/Core/CredentialManager/Infrastructure/SFCredentialManagerWindowsNativeImpl.cs

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Runtime.InteropServices;
+using System.Text;
+using Snowflake.Data.Client;
+using Snowflake.Data.Core.Tools;
+using Snowflake.Data.Log;
+
+namespace Snowflake.Data.Core.CredentialManager.Infrastructure
+{
+
+    internal class SFCredentialManagerWindowsNativeImpl : ISnowflakeCredentialManager
+    {
+        private static readonly SFLogger s_logger = SFLoggerFactory.GetLogger<SFCredentialManagerWindowsNativeImpl>();
+
+        public static readonly SFCredentialManagerWindowsNativeImpl Instance = new SFCredentialManagerWindowsNativeImpl();
+
+        public string GetCredentials(string key)
+        {
+            s_logger.Debug($"Getting the credentials for key: {key}");
+            var hashKey = key.ToSha256Hash();
+            IntPtr nCredPtr;
+            if (!CredRead(hashKey, 1 /* Generic */, 0, out nCredPtr))
+            {
+                s_logger.Info($"Unable to get credentials for key: {key}");
+                return "";
+            }
+
+            var critCred = new CriticalCredentialHandle(nCredPtr);
```

Comment:
class CriticalHandleZeroOrMinusOneIsInvalid implements disposable interface. So I think you should add `using` here.

Reply:
Done, thanks!!!

Reply:
The changes was not push, now you can check it out!!